### DISCUSSION TITLE
No such file or doirectory found with comand: python3 southwest.py CONFIRMATION_NUMBER FIRST_NAME LAST_NAME

### DISCUSSION THREAD

**Discussion by sunshinel0** [jdholtz/auto-southwest-check-in#276] [Q&A] [answered]
so i've downloaded everything to my local mac and have been able to successfully utilize the auto check in. However when I close the terminal and then reopen if I utilize "python3 southwest.py CONFIRMATION_NUMBER FIRST_NAME LAST_NAME" I get back this error: /Library/Frameworks/Python.framework/Versions/3.12/Resources/Python.app/Contents/MacOS/Python: can't open file '/Users/jknapp/southwest.py': [Errno 2] No such file or directory

if I insetad input: git clone https://github.com/jdholtz/auto-southwest-check-in.git
cd auto-southwest-check-in

it will in fact inititalize the auto check in and from there I am able to Successfuly utilize: python3 southwest.py CONFIRMATION_NUMBER FIRST_NAME LAST_NAME

Is there a reason I have to do this prior to every time I wish to initiate the auto check in? any tips would be great I'm very new to this. thank you!

**Comment by jdholtz** [comment]
> Is there a reason I have to do this prior to every time I wish to initiate the auto check in?

When executing the command `python southwest.py ...`, Python will look for a file names `southwest.py` in your current directory. Therefore, you need to be in the `auto-southwest-check-in` directory to do so. You can execute it in any directory using `python /full/path/to/auto-southwest-check-in/southwest.py`.

Testing this, I just found that the logs will be written in whatever directory you execute the command from instead of the `auto-southwest-check-in` directory. I will fix this in the next release.

**Reply by sunshinel0** [reply]
Interesting I tried: python /full/path/to/auto-southwest-check-in/southwest.py

and got this error: zsh: command not found: python

It also did not change me to the correct directory

**Reply by sunshinel0** [reply]
<img width="685" alt="Screenshot 2024-06-29 at 3 53 31 PM" src="https://github.com/jdholtz/auto-southwest-check-in/assets/114316004/0a095b40-da70-48b5-8fdf-a31f5690997c">

**Reply by jdholtz** [reply]
> Interesting I tried: python /full/path/to/auto-southwest-check-in/southwest.py

The first error indicates you should install Python which is a requirement for the script (see the [Prerequisites](https://github.com/jdholtz/auto-southwest-check-in?tab=readme-ov-file#prerequisites)).

Also, you should replace `/full/path/to` with the actual full directory path to the auto-southwest-check-in directory. For example, say you have the `auto-southwest-check-in` directory in your home directory. The command would be `python /home/sunshineI0/auto-southwest-check-in/southwest.py CONF FIRST LAST`.

**Reply by sunshinel0** [reply, marked as answer]
ok, so I didn't realize that 'full/path/to' was a filler for where it actually resides, lol. However, when I attempted the correct path, in this case, `user/jknapp/auto-southwest-check-in` I was still getting an error message. I tried just copying the version command `python3 southwest.py --version` to see its output. I noticed the file path it Automatically assumed was `/Users/jknapp/southwest.py` being the folder was named auto-southwest-check-in. I decided to try renaming the folder to 'southwest.py' This got me a bit further. Still, now it was searching within that folder for `__main__` module.

From there, I tried renaming the script from `southwest.py` to `__main__.py`, and now all of the commands work as intended without the need for me to utilize the git clone command to initialize any subsequent commands.
<img width="1341" alt="image" src="https://github.com/jdholtz/auto-southwest-check-in/assets/114316004/8d20e285-a720-4cf4-87ae-74069654a9aa">

Not sure if this will cause issues with updates in the future, but its the fix that i've landed on for now

**Reply by jdholtz** [reply]
> Not sure if this will cause issues with updates in the future, but its the fix that i've landed on for now

It probably will as the Git history will differ between the next update and your local changes. In the future, I would just recommend to run the script from the directory it resides in or doing python3 /Users/jknapp/auto-southwest-check-in/southwest.py CONF FIRST LAST. Glad you got it working!
